### PR TITLE
FTX widget: default to ftx.com or ftx.us depending on OS region

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/16121

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- Set OS region to Canada
- Start browser (fresh profile)
- Open NTP
- View FTX widget
- Default host should be .com

### CODE REVIEW

File: components/ftx/browser/ftx_pref_utils.cc [hunk 1]
```diff
@@ -5,15 +5,23 @@
 
 #include "brave/components/ftx/browser/ftx_pref_utils.h"
 
+#include <string>
+
 #include "brave/components/ftx/common/pref_names.h"
+#include "brave/components/l10n/browser/locale_helper.h"
```
Comment (simonhong) [inline components/ftx/browser/ftx_pref_utils.cc:11]: maybe we need to add deps for this including?

**Review by simonhong — APPROVED** [review]
LGTM per fixing deps

File: components/ftx/browser/ftx_pref_utils.cc (hunk 1 above)
Comment (petemill) [reply components/ftx/browser/ftx_pref_utils.cc:11]: thanks, fixed

File: components/brave_new_tab_ui/widgets/ftx/components/preOptIn.tsx [hunk 2]
```diff
@@ -32,13 +41,13 @@ export default function PreOptIn (props: Props) {
       <BasicBox isFlex={true} $gap={10} justify='flex-end'>
         <OptionButton
           isSelected={!isDotUsSelected}
-          onClick={setIsDotUsSelected.bind(undefined, false)}
+          onClick={setUserSetIsUSRegion.bind(undefined, false)}
```
Comment (bsclifton) [inline components/brave_new_tab_ui/widgets/ftx/components/preOptIn.tsx:44]: Not necessarily a bug - as the correct default is picked. But (with US region) if I click this one for example (.com), it doesn't save. When I click, it does correctly display below `FTX.com`... and after quit / restarting, it's resetting to `FTX.us`.  Is this OK? Or should it be setting `props.ftx.ftxHost`?

**Review by bsclifton — APPROVED** [review]
Left a comment about pre-opt in behavior not saving; but if that's not a deal-breaker (it's outside the scope of this issue being fixed), we're good to go 😄 👍

File: components/brave_new_tab_ui/widgets/ftx/components/preOptIn.tsx (hunk 2 above)
Comment (petemill) [reply components/brave_new_tab_ui/widgets/ftx/components/preOptIn.tsx:44]: Yes I think it's unnecessary to persist before the user logs in. This is not something that toggles anything until the user connects.

**Comment by srirambv** [comment]
Verification passed on 

Brave | 1.29.49 Chromium: 92.0.4515.131 (Official Build) nightly (64-bit)
-- | --
Revision | `6b8d6c56ce21e38a72f7c4becb5abc1fa5134f29-refs/branch-heads/4515@{#1933}`
OS | Linux

- Verified for non en-US locale (DE) widget defaults to `.com` 

https://user-images.githubusercontent.com/17010094/128182092-34e77123-0e4a-49a2-b626-c05ad58e3315.mov

- Verified for en-US locale widget defaults to `.us` 

https://user-images.githubusercontent.com/17010094/128180527-b95c9e9e-d9ad-4fa5-871a-57d2c18df340.mov

**Pull request merged**